src/Elliptic/Systems/BnsInitialData/BoundaryConditions/StarSurface.hpp

```diff
@@ -0,0 +1,105 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+#include <string>
+#include <vector>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/Tags.hpp"
+#include "Domain/Tags/FaceNormal.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryConditionType.hpp"
+#include "Elliptic/Systems//BnsInitialData/Tags.hpp"
+#include "Options/Context.hpp"
+#include "Options/String.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+/// \endcond
+
+namespace BnsInitialData ::BoundaryConditions {
+
+/*!
+ * Impose StarSurface boundary conditions:
+ * \f[
+ *   n_i F^i = \frac{C}{\alpha^2} B^i n_i.
```

Comment:
Question: Since this is a Neumann boundary condition, it leaves the freedom for a constant offset in `\Phi`. I think the equations also leave this freedom. Have you thought about how to fix this freedom?

Reply:
I haven't, I would think a reasonable choice would be that Phi should be zero at the center of the star or something like that, we probably can't set Phi=0 at the surface because I'm almost sure it's not "equipotential"